src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -105,7 +106,7 @@ def generate_payload(self):
                 props.update(base)
                 props.update(cli)
                 props.update({CORRELATION_ID_PROP_NAME: str(uuid.uuid4()),
-                              'Reserved.EventId': str(uuid.uuid4())})
+                              'Reserved.EventId': _get_or_create_event_id()})
```

Comment:
It not good programming practice to let an external function like `_get_or_create_event_id` set the instance's attribute (`self.event_id`). Consider using a [method](https://docs.python.org/3/tutorial/classes.html#instance-objects) instead.